pypesto/profile/profile.py

```diff
@@ -0,0 +1,254 @@
+import logging
+import numpy as np
+from pypesto import Result
+from ..optimize import OptimizeOptions, minimize
+from .profiler import ProfilerResult
+from .profile_startpoint import fixed_step
+
+logger = logging.getLogger(__name__)
+
+
+class ProfileOptions(dict):
+    """
+    Options for optimization based profiling.
+
+    Parameters
+    ----------
+
+    next_profile_startpoint: function handle, optional
+        function which creates the next startpoint for optimization
+        from the profile from the profile history
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.default_step_size = 0.01
+        self.ratio_min = 0.145
+
+    def __getattr__(self, key):
+        try:
+            return self[key]
+        except KeyError:
+            raise AttributeError(key)
+
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    @staticmethod
+    def assert_instance(maybe_options):
+        """
+        Returns a valid options object.
+
+        Parameters
+        ----------
+
+        maybe_options: OptimizeOptions or dict
+        """
+        if isinstance(maybe_options, ProfileOptions):
+            return maybe_options
+        options = ProfileOptions(**maybe_options)
```

Comment:
i think that to be able to use this, the constructor ``__init__`` above must accept default_step_size and ratio_min as arguments (I think that would make sense anyway).